compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs

```diff
@@ -56,8 +58,10 @@ convertPrim _ "BEGreater" (TBuiltin a1 :-> TBuiltin a2 :-> TBool) | a1 == a2 =
     EBuiltin $ BEGreater a1
 convertPrim _ "BEEqualList" ((a1 :-> a2 :-> TBool) :-> TList a3 :-> TList a4 :-> TBool) | a1 == a2, a2 == a3, a3 == a4 =
     EBuiltin BEEqualList `ETyApp` a1
-convertPrim _ "BEEqualContractId" (TContractId a1 :-> TContractId a2 :-> TBool) | a1 == a2 =
-    EBuiltin BEEqualContractId `ETyApp` a1
+convertPrim v "BEEqualContractId" (TContractId a1 :-> TContractId a2 :-> TBool) | a1 == a2 =
+    if v `supports` featureGenMap
+        then EBuiltin BEEqualGeneric `ETyApp` TContractId a1
```

Comment:
I thought the plan was to keep `BEEqualContractId`, at least that’s what @remyhaemmerle-da  changed the spec to in https://github.com/digital-asset/daml/pull/3812

Reply:
Looking at the code, I don’t really understand why we are not deprecating `BEEqualContractId`. I thought it was something about relative vs absolute contract ids but looking at the code they compile to the same thing in speedy.

Reply:
Let's deprecate it then. (-:

Reply:
Fine with me but @remyhaemmerle-da should take a look before we do this.

Reply:
I fine to deprecate id.